daemon/daemon.go

```diff
@@ -963,8 +963,11 @@ func (daemon *Daemon) initDiscovery(config *Config) error {
 // - Daemon live restore
 func (daemon *Daemon) Reload(config *Config) error {
 	var err error
-	// used to hold reloaded changes
-	attributes := map[string]string{}
+
+	daemon.configStore.reloadLock.Lock()
+	defer daemon.configStore.reloadLock.Unlock()
```

Comment:
Hmm, yeah, I also see comment that `SystemInfo` holds that Lock, but it isn't, which is probably and error.